### PR TITLE
RepositoryCache: add blake3 key type

### PR DESCRIPTION
**Pull request by sluongng**
Support storing repository cache with blake3 hash function.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCache.java [hunk 1]
```diff
@@ -43,7 +44,8 @@ public enum KeyType {
     SHA1("SHA-1", "\\p{XDigit}{40}", "sha1", Hashing.sha1()),
     SHA256("SHA-256", "\\p{XDigit}{64}", "sha256", Hashing.sha256()),
     SHA384("SHA-384", "\\p{XDigit}{96}", "sha384", Hashing.sha384()),
-    SHA512("SHA-512", "\\p{XDigit}{128}", "sha512", Hashing.sha512());
+    SHA512("SHA-512", "\\p{XDigit}{128}", "sha512", Hashing.sha512()),
+    BLAKE3("BLAKE3", "\\p{XDigit}{64}", "blake3", BazelHashFunctions.BLAKE3.getHashFunction());
```
[the remark this answers is not in the document]
Comment (sluongng) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCache.java:48]: Oh good call. 

In a previous iteration I implemented a getInstance static method in Blake3HashFunction. Perhaps I should revert back to that approach instead.

**Comment by sgowroji** [comment]
@sluongng  Could you please take a look at the failing checks?

File: src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCache.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sluongng) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCache.java]: Done

**Comment by sluongng** [comment]
@sgowroji I think there has been some flakiness in the Bazel CI setup. I have checked the old failures as well as the most recent ones and I don't think they are related to my change.

File: src/main/java/com/google/devtools/build/lib/vfs/bazel/Blake3HashFunction.java [hunk 2]
```diff
@@ -25,6 +25,14 @@
 
 /** A {@link HashFunction} for BLAKE3. */
 public final class Blake3HashFunction implements HashFunction {
+  private static Blake3HashFunction blake3HashFunction;
```
Comment (fmeum) [inline src/main/java/com/google/devtools/build/lib/vfs/bazel/Blake3HashFunction.java:28]: Could this just be a static final field initialized inline?

Comment (sluongng) [reply src/main/java/com/google/devtools/build/lib/vfs/bazel/Blake3HashFunction.java]: Oh right, that is indeed much cleaner 👍

**Review by fmeum — APPROVED** [review]

**Comment by fmeum** [comment]
@meteorcloudy Could you review this?

**Review by meteorcloudy — APPROVED** [review]

**Comment by fmeum** [comment]
@bazel-io fork 7.2.0

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.2.0rc1. Thanks!

**Pull request closed without merging**